### DISCUSSION TITLE
GANomaly: AssertionError with my custom data

### DISCUSSION THREAD

**Discussion by ghost** [open-edge-platform/anomalib#1471] [Q&A] [answered]
I have successfully run the segmentation by padim, efficient_ad, and stfpm on my custom dataset. However, when I modify the config file of ganomaly just like the way I did for the others and I run it, it gives me AssertionError during the Testing DataLoader. Below is my modified config.yaml file for ganomaly converted into a text file
[config_ganomaly.txt](https://github.com/openvinotoolkit/anomalib/files/13330684/config_ganomaly.txt)
. The error message is at the 165th line of the post_processing/visualizer.py i.e.,

> assert image_result.pred_mask is not None

**Comment by blaz-r** [comment, marked as answer]
I believe this is the case because ganomaly is classification model and doesn't provide segmentation result.
